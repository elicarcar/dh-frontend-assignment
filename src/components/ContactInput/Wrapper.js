import styled from "styled-components";
const Wrapper = styled.div`
  position: relative;
  display: inline-block;
  margin: 20px;
  &:before {
    display: ${props => (props.displaySearchIcon ? "block" : "none")};
    position: absolute;
    content: " âš² ";
    top: 4px;
    left: 2px;
    height: 10px;
    width: 24px;
    margin-right: 5px;
    transform: rotate(-45deg);
  }

  &:after {
    display: block;
    position: absolute;
    top: 9px;
    right: 9px;
    content: " v ";
  }
`;

export default Wrapper;
