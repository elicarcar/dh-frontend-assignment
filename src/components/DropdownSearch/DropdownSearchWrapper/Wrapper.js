import styled from "styled-components";
const Wrapper = styled.div.attrs(props => ({
  id: "wrapper-div",
  ref: props.divRef,
  displaySearchIcon: props.displaySearchIcon
}))`
  position: relative;
  display: inline-block;
  margin: 20px;
  &:before {
    display: ${props => (props.displaySearchIcon ? "block" : "none")};
    position: absolute;
    content: " ⚲ ";
    transform: scale(0.5);
    top: 4px;
    left: 2px;
    height: 10px;
    width: 24px;
    margin-right: 5px;
    transform: rotate(-45deg);
  }

  &:after {
    content: "▼";
    display: block;
    position: absolute;
    top: 11px;
    right: 45px;
  }
`;

export default Wrapper;
