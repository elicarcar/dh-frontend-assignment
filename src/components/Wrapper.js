import styled from "styled-components";
const Wrapper = styled.div`
  position: relative;
  display: inline-block;
  &:before {
    display: ${props => (props.displaySearchIcon ? "block" : "none")};
    position: absolute;
    content: "âš²";
    top: 4px;
    left: 2px;
    height: 10px;
    width: 24px;
    margin-right: 5px;
    transform: rotate(-45deg);
  }

  &:after {
    display: block;
    position: absolute;
    top: 9px;
    right: 9px;
    content: " 8 ";
  }
`;

export default Wrapper;
