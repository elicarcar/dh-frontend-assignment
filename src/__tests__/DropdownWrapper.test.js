import React from "react";
import { create } from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";
import Wrapper from "../components/DropdownSearch/DropdownSearchWrapper/Wrapper.js";

describe("<Wrapper/>", () => {
  it("has the style rules", () => {
    const tree = create(<Wrapper />).toJSON();
    expect(tree).toHaveStyleRule("position", "relative");
    expect(tree).toHaveStyleRule("display", "inline-block");
    expect(tree).toHaveStyleRule("margin", "20px");
    expect(tree).toHaveStyleRule("position", "absolute", {
      modifier: ":before"
    });
    expect(tree).toHaveStyleRule("display", "none", {
      modifier: ":before"
    });
  });

  it("matches snapshot", () => {
    const tree = create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
