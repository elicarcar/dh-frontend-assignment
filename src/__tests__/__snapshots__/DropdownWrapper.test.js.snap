// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper/> matches snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  margin: 20px;
}

.c0:before {
  display: none;
  position: absolute;
  content: " ⚲ ";
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  top: 4px;
  left: 2px;
  height: 10px;
  width: 24px;
  margin-right: 5px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c0:after {
  content: "▼";
  display: block;
  position: absolute;
  top: 11px;
  right: 45px;
}

<div
  className="c0"
  id="wrapper-div"
/>
`;
